plenoptic/simulate/models/frontend.py

```diff
@@ -1,96 +1,414 @@
+    Model is described in [1]_ and [2]_.
+
+    Parameters
+    ----------
+    kernel_size:
+        Shape of convolutional kernel.
+    on_center:
+        Dictates whether center is on or off; surround will be the opposite of center
+        (i.e. on-off or off-on).
+    width_ratio_limit:
+        Sets a lower bound on the ratio of `surround_std` over `center_std`.
+        The surround Gaussian must be wider than the center Gaussian in order to be a
+        proper Difference of Gaussians. `surround_std` will be clamped to `ratio_limit`
+        times `center_std`.
+    amplitude_ratio:
+        Ratio of center/surround amplitude. Applied before filter normalization.
+    pad_mode:
+        Padding for convolution, defaults to "circular".
+    activation:
+        Activation function following linear convolution.
+
+    Attributes
+    ----------
+    center_surround: nn.Module
+        Difference of Gaussians linear filter.
+    luminance: nn.Module
+        Gaussian convolutional kernel used to normalize signal by local luminance.
+    luminance_scalar: nn.Parameter
+        Scale factor for luminance normalization.
+
+    References
+    ----------
+    .. [1] A Berardino, J Ballé, V Laparra, EP Simoncelli, Eigen-distortions of hierarchical
+        representations, NeurIPS 2017; https://arxiv.org/abs/1710.02266
+    .. [2] http://www.cns.nyu.edu/~lcv/eigendistortions/ModelsIQA.html
+    """
+    def __init__(
+        self,
+        kernel_size: Union[int, Tuple[int, int]],
+        on_center: bool = True,
+        width_ratio_limit: float = 4.0,
+        amplitude_ratio: float = 1.25,
+        pad_mode: str = "circular",
+        activation: Callable[[Tensor], Tensor] = F.softplus,
+    ):
+        super().__init__()
+        self.center_surround = CenterSurround(
+            kernel_size,
+            on_center,
+            width_ratio_limit,
+            amplitude_ratio,
+            pad_mode=pad_mode,
+        )
+        self.luminance = Gaussian(kernel_size=kernel_size)
+        self.luminance_scalar = nn.Parameter(torch.rand(1) * 10)
+        self.activation = activation
+
+    def forward(self, x: Tensor) -> Tensor:
+        linear = self.center_surround(x)
+        lum = self.luminance(x)
+        lum_normed = linear / (1 + self.luminance_scalar * lum)
+        y = self.activation(lum_normed)
+        return y
+
+    def display_filters(self, zoom=5.0, **kwargs):
+        """Displays convolutional filters of model
+
+        Parameters
+        ----------
+        zoom: float
+            Magnification factor for po.imshow()
+        **kwargs:
+            Keyword args for po.imshow
+        Returns
+        -------
+        fig: PyrFigure
         """
+
+        weights = torch.cat(
+            [
+                self.center_surround.filt,
+                self.luminance.filt,
+            ],
+            dim=0,
+        ).detach()
+
+        title = ["linear", "luminance norm",]
```

Comment:
`luminance norm` is a weird title to me (norm makes me think of a scalar, like "l2 norm"). This is the luminance filter used to normalize the output, right? is there a more descriptive name we could give?

Reply:
Fair enough. Changed them to say "linear filt", "luminance filt", "contrast filt"